packages/flutter_tools/test/integration.shard/debug_adapter/flutter_adapter_test.dart

```diff
@@ -99,6 +99,13 @@ void main() {
         '',
         startsWith('Exited'),
       ]);
+
+      // If we're running with an out-of-process debug adapter, ensure that its
+      // own process shuts down after we terminated.
+      final DapTestServer server = dap.server;
+      if (server is OutOfProcessDapTestServer) {
```

Comment:
Will the concrete class of `server` sometimes be something else when this test is run?

Reply:
You can run the tests with an env variable to un them in-process which is useful for debugging (it's faster, you can step through the code, and you don't need to rebuild flutter_tools):

https://github.com/flutter/flutter/blob/78d924af1d1a8841e693892af73d05686c2f7cb7/packages/flutter_tools/test/integration.shard/debug_adapter/test_support.dart#L25

However in that case, there is no `process` (it only exists on `OutOfProcessDapTestServer`).